OpenRA.Game/Network/OrderManager.cs

```diff
@@ -127,6 +128,8 @@ public void TickImmediate()
 						frameData.ClientQuit(clientId, frame);
 					else if (packet.Length >= 5 && packet[4] == (byte)OrderType.SyncHash)
 						CheckSync(packet);
+					else if (packet.Length == 4 + 1 + 8 && packet[4] == (byte)OrderType.WinState)
```

Comment:
You don't need to handle the winstate on the client. In case of a real desync the sync frames will already trigger and in case of a malicious client, the server needs to handle the situation.

Reply:
We should account for `World.PlayerWinStates` in `World.SyncHash()` to guarantee this.

Reply:
I kept the check but with a no-op to prevent re-inserting the packet multiple time. The win/loss state check mechanism is moved server side.